lib/jekyll/commands/serve.rb

```diff
@@ -63,6 +104,76 @@ def process(opts)
           start_up_webrick(opts, destination)
         end
 
+        def shutdown
+          @server.shutdown if running?
+        end
+
+        # Perform logical validation of CLI options
+
+        private
+        def validate_options(opts)
+          if opts["livereload"]
+            if opts["detach"]
+              Jekyll.logger.warn "Warning:",
+                "--detach and --livereload are mutually exclusive. Choosing --livereload"
+              opts["detach"] = false
+            end
+            if opts["ssl_cert"] || opts["ssl_key"]
+              # This is not technically true.  LiveReload works fine over SSL, but
+              # EventMachine's SSL support in Windows requires building the gem's
+              # native extensions against OpenSSL and that proved to be a process
+              # so tedious that expecting users to do it is a non-starter.
+              Jekyll.logger.abort_with "Error:", "LiveReload does not support SSL"
```

Comment:
I've stepped way back from this PR, never had the time to look at it was my plugin was just working for me.  However, Livereload can support SSL, I have support for it in my plugin https://github.com/RobertDeRose/jekyll-livereload/commit/34df507e4dadee09554ca06479de514899b9af39

Reply:
Thanks @RobertDeRose, I was using your plugin until now :)
Are you interested in submitting a patch?

Reply:
I might, if I can find the time.  So, that's a big maybe

Reply:
@RobertDeRose The problem wasn't with LiveReload not supporting SSL; it was with getting EventMachine and OpenSSL to cooperated on Windows.  I actually had it working fine in Linux, but getting the OpenSSL libraries to install and work on the Windows CI environment was a nightmare, so I just gave up 😭.  I figured most people would only be using LiveReload for rapid prototyping so I didn't consider SSL support to be a must-have.

Reply:
@awood that's a fair point, I never did testing for Windows, and to be honest, not a lot of benefit for it when developing locally, unless you really need to ensure SSL isn't messing with thing, but at that point, you don't really need live reload.  OK ignore my extremely late to the party comments then.